onramps.php

```diff
@@ -0,0 +1,136 @@
+
+    if (array_key_exists($brand_plugin, $plugins)) {
+        $add_plugin_placeholder = false;
+    }
+
+    if($add_plugin_placeholder){
+        if("The Bluehost Plugin " === $brand_plugin) {
+            $plugins['bluehost-wordpress-plugin/bluehost-wordpress-plugin.php'] = array(
+                "Name" => "The Bluehost Plugin",
+                "TextDomain"=> "",
+                "Author"    => "Bluehost",
+                "Version" => "",
+                "PluginURI" => "",
+                "AuthorURI" => "",
+                "Description" => ""
+            );
+        }
+        else if("The Hostgator Plugin " === $brand_plugin) {
+            $plugins['wp-plugin-hostgator/wp-plugin-hostgator.php'] = array(
+                "Name" => "The Hostgator Plugin",
+                "TextDomain"=> "",
+                "Author"    => "Hostgator",
+                "Version" => "",
+                "PluginURI" => "",
+                "AuthorURI" => "",
+                "Description" => ""
+            );
+        }
+       
+    }
+    return $plugins;
+}
+
+/**
+ * save brand plugin name on activation
+ */
+function store_brand_plugin_name_on_activation($plugin) {
+    update_brand_plugin_name($plugin);
+}
+
+/**
+ * save brand plugin name on deactivation
+ */
+function store_brand_plugin_name_on_deactivation($plugin) {
+    update_brand_plugin_name($plugin);
+}
+
+function update_brand_plugin_name($plugin) {
+    $current_brand_plugin = get_option('nfd_brand_plugin', '');
+
+    $plugin_data = get_plugin_data(WP_PLUGIN_DIR . '/' . $plugin);
+    $plugin_name = $plugin_data['Name'];
+
+    if ('The Bluehost Plugin' === $plugin_name || 'The Hostgator Plugin' === $plugin_name) {
+        if (empty($current_brand_plugin) || $current_brand_plugin !== $plugin_name) {
+            update_option('nfd_brand_plugin', $plugin_name);
+        }
+    }
+}
+
+/**
+ * On click of activate install and activate the plugin
+ */
+function check_and_install_plugin() {
+    if (isset($_GET['action']) && $_GET['action'] == 'activate' && isset($_GET['plugin']) && $_GET['plugin'] == 'wp-plugin-hostgator/wp-plugin-hostgator.php') {
+        $current_brand_plugin = get_option('nfd_brand_plugin', '');
+        if('The Bluehost Plugin' === $current_brand_plugin) {
+            $plugin_slug = 'bluehost-wordpress-plugin';
+            $plugin_main_file = 'bluehost-wordpress-plugin.php';
+            $plugin_zip_url = 'https://hiive.cloud/workers/release-api/plugins/bluehost/bluehost-wordpress-plugin/download';
+        }
+        else if('The Hostgator Plugin' === $current_brand_plugin) {
+            $plugin_slug = 'wp-plugin-hostgator';
+            $plugin_main_file = 'wp-plugin-hostgator.php';
+            $plugin_zip_url = 'https://hiive.cloud/workers/release-api/plugins/newfold-labs/wp-plugin-hostgator/download';
+        }
+       
+
+        // Download the plugin zip file
+        $tmp_file = download_url($plugin_zip_url);
```

Comment:
What is preventing this code from running when the plugin already exists, but is just being re-activated?

Reply:
If the brand plugin already exists in the plugin list, we will not add a placeholder for it. Therefore, there will be no chance of reactivating the plugin.

Reply:
It seems like it would run either way because it only watches for URL parameter changes. I think the way to test would be to install the Bluehost plugin, deactivate it, add a random file in the plugin, and then activate the plugin with this MU plugin present. If the random file you added disappears, then we have a problem and should look into fixing it. If not, then we're fine.

Reply:
Tested by adding a random file inside the plugin, then deactivating it and activating it again. The random file did not disappear.